<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.34">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Databases Available on Mantis â€“ UCHC HPC</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./img/favicon.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-c8ad9e5dbd60b7b70b38521ab19b7da4.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-21e246651e1b5bb22b0dbf251f7ee916.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar docked quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./databases.html">Databases</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./img/cbc.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./account.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Account &amp; Password</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./connect.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Connecting</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./storage.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">File Systems &amp; Storage</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./running.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Running Jobs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./transfer.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transferring Data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Software</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./databases.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Databases</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./help.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get Help</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#blast" id="toc-blast" class="nav-link active" data-scroll-target="#blast">BLAST</a></li>
  <li><a href="#busco" id="toc-busco" class="nav-link" data-scroll-target="#busco">BUSCO</a></li>
  <li><a href="#cellranger" id="toc-cellranger" class="nav-link" data-scroll-target="#cellranger">CellRanger</a></li>
  <li><a href="#centrifuge" id="toc-centrifuge" class="nav-link" data-scroll-target="#centrifuge">Centrifuge</a></li>
  <li><a href="#dfam" id="toc-dfam" class="nav-link" data-scroll-target="#dfam">Dfam</a></li>
  <li><a href="#diamond" id="toc-diamond" class="nav-link" data-scroll-target="#diamond">DIAMOND</a></li>
  <li><a href="#eggnog" id="toc-eggnog" class="nav-link" data-scroll-target="#eggnog">eggNOG</a></li>
  <li><a href="#entap" id="toc-entap" class="nav-link" data-scroll-target="#entap">EnTAP</a></li>
  <li><a href="#fcs-gx" id="toc-fcs-gx" class="nav-link" data-scroll-target="#fcs-gx">FCS-GX</a></li>
  <li><a href="#gene-ontology-go" id="toc-gene-ontology-go" class="nav-link" data-scroll-target="#gene-ontology-go">Gene Ontology (GO)</a></li>
  <li><a href="#gtdb" id="toc-gtdb" class="nav-link" data-scroll-target="#gtdb">GTDB</a></li>
  <li><a href="#humann" id="toc-humann" class="nav-link" data-scroll-target="#humann">HUMAnN</a></li>
  <li><a href="#kraken" id="toc-kraken" class="nav-link" data-scroll-target="#kraken">Kraken</a></li>
  <li><a href="#long-ranger" id="toc-long-ranger" class="nav-link" data-scroll-target="#long-ranger">Long Ranger</a></li>
  <li><a href="#metaphlan" id="toc-metaphlan" class="nav-link" data-scroll-target="#metaphlan">MetaPHlAn</a></li>
  <li><a href="#mothur" id="toc-mothur" class="nav-link" data-scroll-target="#mothur">Mothur</a></li>
  <li><a href="#orthodb" id="toc-orthodb" class="nav-link" data-scroll-target="#orthodb">OrthoDB</a></li>
  <li><a href="#pfam" id="toc-pfam" class="nav-link" data-scroll-target="#pfam">Pfam</a></li>
  <li><a href="#recentrifuge" id="toc-recentrifuge" class="nav-link" data-scroll-target="#recentrifuge">Recentrifuge</a></li>
  <li><a href="#refseq-genomes" id="toc-refseq-genomes" class="nav-link" data-scroll-target="#refseq-genomes">RefSeq Genomes</a></li>
  <li><a href="#snpeff" id="toc-snpeff" class="nav-link" data-scroll-target="#snpeff">SnpEff</a></li>
  <li><a href="#sortmerna" id="toc-sortmerna" class="nav-link" data-scroll-target="#sortmerna">SortMeRNA</a></li>
  <li><a href="#trinotate" id="toc-trinotate" class="nav-link" data-scroll-target="#trinotate">Trinotate</a></li>
  <li><a href="#uniprot" id="toc-uniprot" class="nav-link" data-scroll-target="#uniprot">UniProt</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Databases Available on Mantis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>We host several commonly used bioinformatics databases on the Mantis cluster. We update these regularly â€“ every three months for databases sourced from NCBI, EMBL-EBI, and SIB repositories and every six months for all other databases. We aim to host the latest version and the previous version of each database at all times. We also host several user-made databases where there is demand.</p>
<section id="blast" class="level2">
<h2 class="anchored" data-anchor-id="blast">BLAST</h2>
<p>The latest versions of the official NCBI BLAST databases are available on Mantis. This directory also includes custom databases constructed from the latest eggNOG, OrthoDB, and UniProt releases.</p>
<section id="example-usage" class="level4">
<h4 class="anchored" data-anchor-id="example-usage">Example usage:</h4>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">blastp</span> <span class="at">-db</span> /isg/shared/databases/blast/v5/nr <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/blast/v4/</li>
<li>/isg/shared/databases/blast/v5/</li>
<li>/isg/shared/databases/blast/eggnog/v5.0.1/</li>
<li>/isg/shared/databases/blast/eggnog/v5.0.2/</li>
<li>/isg/shared/databases/blast/orthodb/v12v1/</li>
<li>/isg/shared/databases/blast/uniprot/v2025.06/</li>
</ul>
</section>
</section>
<section id="busco" class="level2">
<h2 class="anchored" data-anchor-id="busco">BUSCO</h2>
<p>The latest versions of the EZLab BUSCO databases are available on Mantis. This directory includes all lineage-specific odb10 and odb12 databases.</p>
<section id="example-usage-1" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-1">Example usage:</h4>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">busco</span> <span class="at">-l</span> /isg/shared/databases/busco/odb12/insecta_odb12/ <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-1" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-1">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/busco/odb10/</li>
<li>/isg/shared/databases/busco/odb12/</li>
</ul>
</section>
</section>
<section id="cellranger" class="level2">
<h2 class="anchored" data-anchor-id="cellranger">CellRanger</h2>
<p>The latest versions of 10X Genomicsâ€™ CellRanger databases are available on Mantis. This directory contains the official human, mouse, and rat genomic and transcriptomic references.</p>
<section id="example-usage-2" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-2">Example usage:</h4>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cellranger</span> count <span class="at">--transcriptome</span> /isg/shared/databases/cellranger/v2024A/refdata-gex-GRCm39-2024-A/ <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-2" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-2">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/cellranger/v2020A/</li>
<li>/isg/shared/databases/cellranger/v2024A/</li>
</ul>
</section>
</section>
<section id="centrifuge" class="level2">
<h2 class="anchored" data-anchor-id="centrifuge">Centrifuge</h2>
<p>The latest official Centrifuge databases, as well as several custom databases, are available on Mantis. This directory contains databases generated from bacterial, archaean, human, viral, and other data, as well as a database generated from NCBI-sourced non-redundant nucleotide sequences.</p>
<section id="example-usage-3" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-3">Example usage:</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">centrifuge</span> <span class="at">-q</span> <span class="at">-x</span> /isg/shared/databases/centrifuge/v2018.04/refseq_b+a+v+h/p+h+v <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-3" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-3">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/centrifuge/custom/</li>
<li>/isg/shared/databases/centrifuge/v2018.04/</li>
</ul>
</section>
</section>
<section id="dfam" class="level2">
<h2 class="anchored" data-anchor-id="dfam">Dfam</h2>
<p>The latest Dfam FamDB databases are available on Mantis. This directory contains all lineage-specific partitions in HDF5 format.</p>
<section id="example-usage-4" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-4">Example usage:</h4>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">famdb.py</span> <span class="at">-i</span> /isg/shared/databases/dfam/v3.9/ <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-4" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-4">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/dfam/v3.8/</li>
<li>/isg/shared/databases/dfam/v3.9/</li>
</ul>
</section>
</section>
<section id="diamond" class="level2">
<h2 class="anchored" data-anchor-id="diamond">DIAMOND</h2>
<p>Several DIAMOND databases are available on Mantis. This directory contains the official eggNOG diamond databases, the BLAST v4 and v5 nr databases in DIAMOND format, and custom databases generated from OrthoDB, RefSeq, and UniProt protein data.</p>
<section id="example-usage-5" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-5">Example usage:</h4>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">diamond</span> blastp <span class="at">-d</span> /isg/shared/databases/diamond/nr/v5/nr.dmnd <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-5" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-5">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/diamond/eggnog/v5.0.1/</li>
<li>/isg/shared/databases/diamond/eggnog/v5.0.2/</li>
<li>/isg/shared/databases/diamond/nr/v4/</li>
<li>/isg/shared/databases/diamond/nr/v5/</li>
<li>/isg/shared/databases/diamond/orthodb/v12v1/</li>
<li>/isg/shared/databases/diamond/refseq/release_230/</li>
<li>/isg/shared/databases/diamond/uniprot/v2025.06/</li>
</ul>
</section>
</section>
<section id="eggnog" class="level2">
<h2 class="anchored" data-anchor-id="eggnog">eggNOG</h2>
<p>The latest eggNOG (emapper) databases are available on Mantis. This directory contains the eggNOG SQL database, DIAMOND database, mmseqs database, and Pfam-A HMM database.</p>
<section id="example-usage-6" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-6">Example usage:</h4>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">emapper.py</span> <span class="at">--mmseqs_db</span> /isg/shared/databases/eggnog/v5.0.1/mmseqs/mmseqs.db <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-6" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-6">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/eggnog/v5.0.1/</li>
<li>/isg/shared/databases/eggnog/v5.0.2/</li>
</ul>
</section>
</section>
<section id="entap" class="level2">
<h2 class="anchored" data-anchor-id="entap">EnTAP</h2>
<p>The latest EnTAP databases are available on Mantis. This directory contains EnTAPâ€™s binary and SQL databases. Other databases commonly used with EnTAP (e.g., assorted DIAMOND databases) may be found in their respective directories.</p>
<section id="example-usage-7" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-7">Example usage:</h4>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">EnTAP</span> <span class="at">--entap-db-sql</span> /isg/shared/databases/entap/v2.1.0/entap_database.db <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-7" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-7">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/entap/v1.0.0</li>
<li>/isg/shared/databases/entap/v2.1.0</li>
</ul>
</section>
</section>
<section id="fcs-gx" class="level2">
<h2 class="anchored" data-anchor-id="fcs-gx">FCS-GX</h2>
<p>The latest FCS-GX databases are available on Mantis. This directory contains the sequence file, index file, and full set of metadata files from each release.</p>
<section id="example-usage-8" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-8">Example usage:</h4>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">python3</span> fcs.py screen genome <span class="at">--gx-db</span> /isg/shared/databases/fcs-gx/v2023.01.24/ <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-8" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-8">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/fcs-gx/v2022.07.08/</li>
<li>/isg/shared/databases/fcs-gx/v2023.01.24/</li>
</ul>
</section>
</section>
<section id="gene-ontology-go" class="level2">
<h2 class="anchored" data-anchor-id="gene-ontology-go">Gene Ontology (GO)</h2>
<p>The latest Gene Ontology databases, as well as several external ontology-to-gene ontology databases, are available on Mantis. This directory contains the full go.obo and go-basic.obo files as well as several files linking external ontological terms to gene ontological terms (e.g., interpro2go, pfam2go, reactome2go).</p>
<section id="usage-note" class="level4">
<h4 class="anchored" data-anchor-id="usage-note">Usage note:</h4>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">These</span> data are general-purpose and may be interfaced with using a variety of software.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-9" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-9">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/gene_ontology/v06.2025/</li>
</ul>
</section>
</section>
<section id="gtdb" class="level2">
<h2 class="anchored" data-anchor-id="gtdb">GTDB</h2>
<p>The latest Genome Taxonomy (GT) databases are available on Mantis. This directory contains all GTDB taxonomy, reference tree, metadata files, and marker genes for all archaean and bacterial GTDB genomes. It also contains the full set of 16S rRNA sequences identified across all high-quality GTDB genomes.</p>
<section id="usage-note-1" class="level4">
<h4 class="anchored" data-anchor-id="usage-note-1">Usage note:</h4>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">These</span> data are general-purpose and may be interfaced with using a variety of software.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-10" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-10">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/gtdb/v220.0/</li>
<li>/isg/shared/databases/gtdb/v226.0/</li>
</ul>
</section>
</section>
<section id="humann" class="level2">
<h2 class="anchored" data-anchor-id="humann">HUMAnN</h2>
<p>The latest HUMAnN databases compatible with our HUMAnN modules are available on Mantis. This directory contains the annotated UniRef50 and UniRef90 databases, filtered UniRef50 and UniRef90 bases, and the full chochophlan database.</p>
<section id="example-usage-9" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-9">Example usage:</h4>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="ex">humann</span> <span class="at">--protein-database</span> /isg/shared/databases/humann/v3/uniref90_annotated/uniref90_201901b_full.dmnd <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-11" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-11">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/humann/v2/</li>
<li>/isg/shared/databases/humann/v3/</li>
</ul>
</section>
</section>
<section id="kraken" class="level2">
<h2 class="anchored" data-anchor-id="kraken">Kraken</h2>
<p>The latest Kraken2 databases are available on Mantis. This directory contains a custom archaean+bacterial+viral+fungal Kraken2 database as well as several official Kraken/Kraken2 databases (e.g., standard, pluspf_16gb, pluspfp_16gb, core_nt).</p>
<section id="example-usage-10" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-10">Example usage:</h4>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2</span> <span class="at">--db</span> /isg/shared/databases/kraken/v06.2025/standard/ <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-12" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-12">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/kraken/custom/</li>
<li>/isg/shared/databases/kraken/v06.2025/</li>
</ul>
</section>
</section>
<section id="long-ranger" class="level2">
<h2 class="anchored" data-anchor-id="long-ranger">Long Ranger</h2>
<p>The latest versions of 10X Genomicsâ€™ Long Ranger databases are available on Mantis. This directory contains the Long Ranger-compatible hg19 (GRCh37), hg38 (GRCh38), and b37/1000 Genomes references.</p>
<section id="example-usage-11" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-11">Example usage:</h4>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="ex">longranger</span> <span class="at">--reference</span><span class="op">=</span>/isg/shared/databases/longranger/v2.1.0/refdata-GRCh38-2.1.0/ <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-13" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-13">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/longranger/v2.1.0/</li>
</ul>
</section>
</section>
<section id="metaphlan" class="level2">
<h2 class="anchored" data-anchor-id="metaphlan">MetaPHlAn</h2>
<p>The latest MetaPHlAn databases compatible with our MetaPHlAn modules are available on Mantis. This directory contains the official Bowtie2 databases and various metadata files.</p>
<section id="example-usage-12" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-12">Example usage:</h4>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="ex">metaphlan</span> <span class="at">--bowtie2db</span> /isg/shared/databases/metaphlan/vOct22/ <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-14" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-14">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/metaphlan/v30/</li>
<li>/isg/shared/databases/metaphlan/vOct22/</li>
</ul>
</section>
</section>
<section id="mothur" class="level2">
<h2 class="anchored" data-anchor-id="mothur">Mothur</h2>
<p>The latest Mothur databases are available on Mantis. This directory contains the official full-length sequence and taxonomy databases (nr), seed databases (seed), and chimera.slayer-compatible silva alignment (gold).</p>
<section id="example-usage-13" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-13">Example usage:</h4>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="ex">align.seqs</span><span class="er">(</span><span class="va">reference</span><span class="op">=</span>/isg/shared/databases/mothur/v138.2/silva.nr_v138_2.align, <span class="op">&lt;</span>other <span class="ex">options</span><span class="op">&gt;</span><span class="kw">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-15" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-15">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/mothur/v132/</li>
<li>/isg/shared/databases/mothur/v138.2/</li>
</ul>
</section>
</section>
<section id="orthodb" class="level2">
<h2 class="anchored" data-anchor-id="orthodb">OrthoDB</h2>
<p>The latest OrthoDB databases are available on Mantis. This directory contains OrthoDB amino acid and coding sequence data as well as several metadata files that describe the relationships between species, sequences, orthogroups, ontological terms, etc.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>Please do not copy these files in their entirety. If you need data for a particular species (e.g., <em>Mus musculus</em> [ID:10090_1]), we recommend using seqkit to extract those sequences and write them to a new file.</p>
</div>
</div>
<section id="example-usage-14" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-14">Example usage:</h4>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="va">CDS</span><span class="op">=</span><span class="st">"/isg/shared/databases/orthodb/v12v1/odb12v1_cds_fasta"</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> grep <span class="at">-pr</span> <span class="st">"10090_1"</span> <span class="st">"</span><span class="va">${CDS}</span><span class="st">"</span> <span class="op">&gt;</span> mmusculus.fna</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-16" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-16">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/orthodb/v12v1/</li>
</ul>
</section>
</section>
<section id="pfam" class="level2">
<h2 class="anchored" data-anchor-id="pfam">Pfam</h2>
<p>The latest Pfam-A databases are available on Mantis. This directory contains the official Pfam-A FASTA and HMM files as well as metadata and alignments in Stockholm format.</p>
<section id="example-usage-15" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-15">Example usage:</h4>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="ex">These</span> data are general-purpose and may be interfaced with using a variety of software.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-17" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-17">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/pfam/v37.3/</li>
<li>/isg/shared/databases/pfam/v37.4/</li>
</ul>
</section>
</section>
<section id="recentrifuge" class="level2">
<h2 class="anchored" data-anchor-id="recentrifuge">Recentrifuge</h2>
<p>The latest Recentrifuge databases are available on Mantis. This directory contains the full set of NCBI Taxonomy taxdump files, including the nodes.dmp and names.dmp files required by Recentrifuge.</p>
<section id="example-usage-16" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-16">Example usage:</h4>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">rcf</span> <span class="at">-n</span> /isg/shared/databases/recentrifuge/v2025.06/taxdump/ <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="paths-to-available-databases-18" class="level5">
<h5 class="anchored" data-anchor-id="paths-to-available-databases-18">Paths to available databases:</h5>
<ul>
<li>/isg/shared/databases/recentrifuge/v2025.06/</li>
</ul>
</section>
</section>
</section>
<section id="refseq-genomes" class="level2">
<h2 class="anchored" data-anchor-id="refseq-genomes">RefSeq Genomes</h2>
<p>The latest genomic data for most major RefSeq Genome lineages are available on Mantis. This directory contains one FASTA file per lineage containing all genomic data for the species in said lineages.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>Please do not copy these files in their entirety. If you need data for a particular species (e.g., <em>Drosophila melanogaster</em>), we recommend using seqkit to extract those sequences and write them to a new file.</p>
</div>
</div>
<section id="example-usage-17" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-17">Example usage:</h4>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="va">INVERTS</span><span class="op">=</span><span class="st">"/isg/shared/databases/refseq_genomes/release_230/invertebrate/invertebrate.genomic.230.fna"</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> grep <span class="at">-npr</span> <span class="st">"Drosophila melanogaster"</span> <span class="st">"</span><span class="va">${INVERTS}</span><span class="st">"</span> <span class="op">&gt;</span> dmel.fna</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-19" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-19">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/refseq_genomes/release_230/archaea/</li>
<li>/isg/shared/databases/refseq_genomes/release_230/bacteria/</li>
<li>/isg/shared/databases/refseq_genomes/release_230/fungi/</li>
<li>/isg/shared/databases/refseq_genomes/release_230/invertebrate/</li>
<li>/isg/shared/databases/refseq_genomes/release_230/plant/</li>
<li>/isg/shared/databases/refseq_genomes/release_230/protozoa/</li>
<li>/isg/shared/databases/refseq_genomes/release_230/vertebrate_mammalian/</li>
<li>/isg/shared/databases/refseq_genomes/release_230/vertebrate_other/</li>
<li>/isg/shared/databases/refseq_genomes/release_230/viral/</li>
</ul>
</section>
</section>
<section id="snpeff" class="level2">
<h2 class="anchored" data-anchor-id="snpeff">SnpEff</h2>
<p>The latest pre-built hg19 and hg38 SnpEff databases are available on Mantis.</p>
<section id="example-usage-18" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-18">Example usage:</h4>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-jar</span> snpEff.jar <span class="at">-dataDir</span> /isg/shared/databases/snpeff/v4.3/hg38/ <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-20" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-20">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/snpeff/v4.3/</li>
</ul>
</section>
</section>
<section id="sortmerna" class="level2">
<h2 class="anchored" data-anchor-id="sortmerna">SortMeRNA</h2>
<p>The latest SortMeRNA databases are available on Mantis. This directory contains the fast, default, and sensitive / sensitive_rfam_seeds rRNA reference data in FASTA format.</p>
<section id="example-usage-19" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-19">Example usage:</h4>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="ex">sortmerna</span> <span class="at">-ref</span> /isg/shared/databases/sortmerna/v4.3.4/smr_v4.3_default_db.fasta <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-21" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-21">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/sortmerna/v4.3.3/</li>
<li>/isg/shared/databases/sortmerna/v4.3.4/</li>
</ul>
</section>
</section>
<section id="trinotate" class="level2">
<h2 class="anchored" data-anchor-id="trinotate">Trinotate</h2>
<p>The latest Trinotate databases compatible with our Trinotate modules are available on Mantis. This directory contains the Trinotate SQLite database, as well as the Pfam-A HMMER and UniProt/Swiss-Prot BLAST databases released around the time of Trinotate v3.2.1.</p>
<section id="example-usage-20" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-20">Example usage:</h4>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> /isg/shared/databases/trinotate/v3.2.1/Trinotate.sqlite myTrinotate.sqlite</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="ex">Trinotate</span> myTrinotate.sqlite <span class="op">&lt;</span>other options<span class="op">&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-22" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-22">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/trinotate/v3.2.1/</li>
</ul>
</section>
</section>
<section id="uniprot" class="level2">
<h2 class="anchored" data-anchor-id="uniprot">UniProt</h2>
<p>A set of the latest UniProt databases are available on Mantis. This directory contains the UniProt/SwissProt, UniRef50, and UniRef90 protein data in FASTA format.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>Please do not copy these files in their entirety. If you need data for a particular species (e.g., <em>Homo sapiens</em>), we recommend using seqkit to extract those sequences and write them to a new file.</p>
</div>
</div>
<section id="example-usage-21" class="level4">
<h4 class="anchored" data-anchor-id="example-usage-21">Example usage:</h4>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="va">UNIREF</span><span class="op">=</span><span class="st">"/isg/shared/databases/uniprot/v2025.06/uniref90.fasta"</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> grep <span class="at">-nrp</span> <span class="st">"Homo sapiens"</span> <span class="st">"</span><span class="va">${UNIREF}</span><span class="st">"</span> <span class="op">&gt;</span> hsapiens.fna</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="paths-to-available-databases-23" class="level4">
<h4 class="anchored" data-anchor-id="paths-to-available-databases-23">Paths to available databases:</h4>
<ul>
<li>/isg/shared/databases/uniprot/v2025.06/</li>
</ul>
<!-- <button type="button" class="btn btn-primary btn-sm chevron-toggle" data-bs-toggle="collapse" 
    data-bs-target="#demo0"> 
  Show Paths <i class="bi bi-chevron-down"></i>
</button> -->
<!-- <div id="demo0" class="collapse">
<ul>
  <li>/isg/shared/databases/blast/v5</li>
  <li>/isg/shared/databases/blast/v4</li>
  <li>/isg/shared/databases/blast/eggnog</li>
</ul>
<hr></hr>

</div> -->
<!-- ### Database 2
Put a brief description of the database here and maybe some instructions about how it is used.

<button type="button" class="btn btn-primary btn-sm chevron-toggle" data-bs-toggle="collapse" 
    data-bs-target="#demo1">
  Show Paths <i class="bi bi-chevron-down"></i>
</button>
<div id="demo1" class="collapse">

<table class="table caption-top" data-quarto-postprocess="true">
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">Col1</td>
<td data-quarto-table-cell-role="th">Col2</td>
</tr>
<tr class="even">
<td>Path</td>
<td>Source</td>
</tr>
<tr class="odd">
<td>Path</td>
<td>Source</td>
</tr>
</tbody>
</table>

</div> -->


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "î§‹";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/CBC-UCONN\.github\.io\/hpc-docs\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>